### PR TITLE
Feature/user coupon

### PR DESCRIPTION
**Pull request by minza55113151**
Pls toi me bab berm berm

### CODE REVIEW

File: src/modules/user/layout/components/UserTopNav.tsx [hunk 1]
```diff
@@ -23,8 +19,8 @@ const UserTopNav = () => {
       <StyledButton
         type="default"
         shape="circle"
-        icon={<User size={22} color={colorPrimary} weight="duotone" />}
-      />
+        icon={<UserAvatar />}
+      ></StyledButton>
```
Comment (JiMeow) [inline src/modules/user/layout/components/UserTopNav.tsx:23]: use self close tag

**Review by lazykern — CHANGES_REQUESTED** [review]
Please make sure to retain the size for smartphones, like the home page

### Orders page (w/maximum width)
localhost_3000_orders

### Coupon page (w/o maximum width)
localhost_3000_orders (1)

File: src/modules/user/coupon/CouponModal.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        }
+      }}
+      onCancel={() => setModalOpen(false)}
+    >
+      <Text>
+        {modalType === "headTable"
+          ? `คุณต้องการเป็นเจ้าของบิลแทน ${headTableName} หรือไม่โดยคูปองที่ ${headTableName} ใช้ไว้จะถูกคืนแต้มทั้งหมด`
+          : "ระบบจะคืนแต้มที่คุณใช้แลกคูปองอันเก่าและใช้งานคูปองใบนี้แทน"}
+      </Text>
+    </Modal>
```
Comment (lazykern) [inline src/modules/user/coupon/CouponModal.tsx:44]: No america allowed

### Current
image

### Should be
image

**Review by lazykern — CHANGES_REQUESTED** [review]

File: src/modules/user/coupon/CouponModal.tsx (hunk 2 above)
Comment (minza55113151) [reply src/modules/user/coupon/CouponModal.tsx:44]: Thx. Done

File: src/modules/user/layout/components/UserTopNav.tsx (hunk 1 above)
Comment (minza55113151) [reply src/modules/user/layout/components/UserTopNav.tsx:23]: Thx. Done

File: src/modules/user/coupon/CouponDrawerContent.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import { H3, H5, Text } from "@/modules/common/components/Typography";
+import { commaFormat } from "@/modules/common/utils";
+import { type Coupon } from "@/modules/user/mock/coupons";
+import styled from "@emotion/styled";
+import { Button } from "antd";
+import Image from "next/image";
+
+type CouponDrawerContentProps = {
+  coupon: Coupon | undefined | null;
```
Comment (JiMeow) [inline src/modules/user/coupon/CouponDrawerContent.tsx:9]: Can we kust use `coupon: Coupon | undefined;`

File: src/modules/user/coupon/CouponModal.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            ? "ต้องการเป็นเจ้าของบิลแทนหรือไม่"
+            : "ใช้คูปองใบนี้แทน?"}
+        </H5>
+      }
+      centered
+      open={modalOpen}
+      onOk={() => {
+        setModalOpen(false);
+        if (modalType === "headTable") {
+          // TODO: change head table by BE
```
Comment (JiMeow) [inline src/modules/user/coupon/CouponModal.tsx:32]: What is BE ?

File: src/modules/user/coupon/OrderCoupon.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            <UserAvatar image={session.headTableUser?.image} />
+          ) : (
+            <UserAvatar image={session.user?.image} />
+          )}
+          <FlexBetweenCol>
+            {couponInUsed ? (
+              <CouponTitle>
+                {'"'}
+                {coupon?.title}
+                {'"'}
```
Comment (JiMeow) [inline src/modules/user/coupon/OrderCoupon.tsx:42]: remove `{'"'}`

File: src/modules/user/coupon/OrderCoupon.tsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+  return (
+    <OrderCouponContainer onClick={onClickOrderCoupon}>
+      <StyledCard>
+        <OrderCouponContent>
+          {!isHeadTable ? (
+            <UserAvatar image={session.headTableUser?.image} />
+          ) : (
+            <UserAvatar image={session.user?.image} />
+          )}
```
Comment (JiMeow) [inline src/modules/user/coupon/OrderCoupon.tsx:36]: I'm not sure but could we use just 
`<UserAvatar image={session.headTableUser?.image} />`

File: src/modules/user/mock/coupons.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
+export type Coupon = {
+  id: string;
+  title: string;
+  image: string;
+  description: string;
+  required_menus: number[];
```
Comment (JiMeow) [inline src/modules/user/mock/coupons.ts:6]: not necessarily but change from `snake case` -> `camel case`

File: src/modules/user/mock/session.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { user, type User } from "@/modules/user/mock/user";
+
+export type Session = {
+  user: User | null;
+  headTableUser: User | null;
+  coupon: string | null;
+  tableNumber: number | null;
```
Comment (JiMeow) [inline src/modules/user/mock/session.ts:7]: Should we use `null` or `undefined`

Comment (JiMeow) [reply src/modules/user/mock/session.ts:7]: Or could we just use ? like 
`talbeNumber?: number;`

File: src/modules/user/mock/user.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
+export type User = {
+  id: string;
+  name: string;
+  image: string;
```
Comment (JiMeow) [inline src/modules/user/mock/user.ts:4]: May be use `image?: string;`

**Review by JiMeow — CHANGES_REQUESTED** [review]
thankyou for being hardwork person 🙏🙏

**Comment by minza55113151** [comment]
> Please make sure to retain the size for smartphones, like the home page
> 
> ### Orders page (w/maximum width)
> localhost_3000_orders
> 
> ### Coupon page (w/o maximum width)
> localhost_3000_orders (1)

Done

File: src/modules/user/coupon/CouponModal.tsx (hunk 4 above)
Comment (minza55113151) [reply src/modules/user/coupon/CouponModal.tsx:32]: Backend

File: src/modules/user/coupon/OrderCoupon.tsx (hunk 5 above)
Comment (minza55113151) [reply src/modules/user/coupon/OrderCoupon.tsx:42]: Can't

File: src/modules/user/coupon/CouponModal.tsx (hunk 4 above)
Comment (minza55113151) [reply src/modules/user/coupon/CouponModal.tsx:32]: Backend

File: src/modules/user/coupon/OrderCoupon.tsx (hunk 6 above)
Comment (minza55113151) [reply src/modules/user/coupon/OrderCoupon.tsx:36]: Done

File: src/modules/user/coupon/CouponDrawerContent.tsx (hunk 3 above)
Comment (minza55113151) [reply src/modules/user/coupon/CouponDrawerContent.tsx:9]: Thx. Done.

File: src/modules/user/mock/user.ts (hunk 9 above)
Comment (minza55113151) [reply src/modules/user/mock/user.ts:4]: I think image will not be null it will be "" ?

**Comment by lazykern** [comment]
Please click resolve conversation after you've done resolving the changes requested

**Comment by minza55113151** [comment]
> Please click resolve conversation after you've done resolving the changes requested
> 
> FYI this is the button i'm referring to image

Done

**Comment by lazykern** [comment]
Everything is looking good now I guess, please fuck with the merge conflict and I will reconfirm the pr status

**Review by lazykern — APPROVED** [review]

**Review by JiMeow — DISMISSED** [review]
I didn't look anything but i wont block this pr so i approve

**Comment by lazykern** [comment]
I didn't look anything but i wont block this pr so i approve

File: src/modules/user/user/UserAvatar.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+      <Avatar
+        size={44}
+        style={{ backgroundColor: colorPrimaryBg }}
+        icon={<User color={colorPrimary} weight="duotone" />}
+      />
+    );
+  }
+};
+
+export default UserAvatar;
```
Comment (GolfGrab) [inline src/modules/user/user/UserAvatar.tsx:26]: fix logic 
avatar to letter type Avatar,color base on user id
uidInt = parseInt(userId, 16)

colors = [ pick some color from this
https://tailwindcss.com/docs/customizing-colors]

usercolor = colors[uidInt % lenColors]

**Review by GolfGrab — CHANGES_REQUESTED** [review]
good work!
have some request changes

**Review by GolfGrab — APPROVED** [review]

**Pull request merged**